packages/node/src/indexer/worker/worker.service.ts

```diff
@@ -17,18 +18,27 @@ export type ProcessBlockResponse = {
   dynamicDsCreated: boolean;
 };
 
+export type WorkerStatusResponse = {
+  threadId: number;
+  isIndexing: boolean;
+  fetchedBlocks: number;
+  toFetchBlocks: number;
+};
+
 @Injectable()
 export class WorkerService {
   private fetchedBlocks: Record<string, BlockContent> = {};
   private currentRuntimeVersion: RuntimeVersion | undefined;
+  private _isIndexing = false;
 
   private queue: AutoQueue<FetchBlockResponse>;
 
   constructor(
     private apiService: ApiService,
     private indexerManager: IndexerManager,
+    private nodeConfig: NodeConfig,
   ) {
-    this.queue = new AutoQueue(undefined, 5);
+    this.queue = new AutoQueue(undefined, nodeConfig.batchSize);
```

Comment:
improve fetch performance for worker (same as before)

Reply:
I found this change made the biggest difference. Even on my branch this change increases the performance to match.